book_price/models/sale_order_line.py

```diff
@@ -0,0 +1,16 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    pricelist_price_unit = fields.Float(string="Book Price", compute='_compute_pricelist_price_unit')
+
+    @api.depends('product_id', 'product_uom', 'product_uom_qty')
+    def _compute_pricelist_price_unit(self):
+        for line in self:
+            if line.product_id:
+                line.pricelist_price_unit = line._get_pricelist_price()
+            else:
+                line.pricelist_price_unit = 0
```

Comment:
```suggestion
                line.pricelist_price_unit = 0.0
```
This approach might helps avoid any potential type-related issues or inconsistencies in your code. :thinking: